hal/src/hal-filesystem.ads

```diff
@@ -92,17 +120,26 @@ package HAL.Filesystem is
    function Read (This : in out File_Handle;
                   Data : out Byte_Array)
                   return Status_Kind is abstract;
+   --  Read the next Data'Length bytes in This and put to in Data. If there
+   --  isn't enough byte to read to fill Data, return Input_Output_Error.
 
    function Write (This : in out File_Handle;
                    Data : Byte_Array)
                    return Status_Kind is abstract;
+   --  Write bytes in Data to This. This overrides the next Data'Length bytes in
+   --  this file, if they exist, it extends the file otherwise.
 
    function Seek (This   : in out File_Handle;
                   Offset : IO_Count)
                   return Status_Kind is abstract;
+   --  Set the read/write cursor of This to point to its byte at the absolute
+   --  Offset.
+   --
+   --  ??? What should happen if this offset is beyond the end of the file?
```

Comment:
ah, I didn't know POSIX lseek could change the size of the file... I guess something closer to UNIX lseek would be less confusing, with SEEK_SET, SEEK_CUR, SEEK_END.